# IIRN_metabolomics
This is the code associated with the paper :

Integrative analysis of serum and fecal metabolome and the microbiome that herald Crohn Disease flare

Metabolites are key mediators of many cellular functions, but scant data describe their temporal dynamics along Crohn Disease (CD) course.
We analyzed serum and fecal metabolomes in a prospective cohort of quiescent CD patients over two years or until the occurrence of a clinical flare, aiming to identify metabolic changes that predict flare. We further defined complex interactions between serum and fecal metabolites, and the gut microbiome. 
We indicate metabolites with persistent signals across comparisons; in CD vs. controls, active CD (clinical/biomarker) vs. those in remission, and in pre-flare samples between relapsers and non-relapsers, which serves as internal validation of the metabolomics signal and indicate metabolitesâ€™ role in CD pathogenesis. Paired multi-omics analyses facilitated the detection of complex interactions that go beyond a single dataset. It indicated an overall inverse shift in energy source in CD (vs. controls) between fat and sugars in the serum and stool. Furthermore, it highlighted interactions between different sugars and oral bacteria mislocated to the CD gut, and between arachidonic-acid unsaturated fat derivatives and CD-associated bacteria including R. gnavus and Fusobacteria taxa. Using pre-flare samples', we managed to predict a subsequent flare using metabolic signatures. This also led to the development of an index based on clinical lab [UA/Cr ratio+log2(CRP)] to predict flare in another, similarly designed, independent cohort.
Our results suggest a potential contribution of the microbial and metabolite composition to CD flare pathogenesis. These factors may guide personalized preemptive therapy intensification and provide insight into the underlying mechanism of CD flares.

The functions used through the code should all be available under the Functions/ directory.
Metadata and normalized data files are under the Datasets/ directory.

Additional run guides are available as readme.txt files under the appropriate subdirectories,
as necessary. 

For the code of the multi-omic analysis performed by the Borenstein lab, please visit their
GitHub Repository.

Data is publicly available at the following:
The 16S amplicon sequencing dataset was deposited as BioProject PRJNA1053872.
Microbial shotgun sequencing is available as BioProject XX.

R session info:
